### PR TITLE
KAFKA-8142: Fix NPE for nulls in Headers

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java [hunk 1]
```diff
@@ -90,7 +90,10 @@ public long sizeBytes() {
         if (headers != null) {
             for (final Header header : headers) {
                 size += header.key().toCharArray().length;
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java:92]: Add a check here that `header.key()` doesn't return `null`?

**Review by bbejeck — APPROVED** [review]
Thanks for fixing this @mjsax. One minor comment but overall LGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java (hunk 1 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java:92]: The key cannot be `null` -- compare https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/header/internals/RecordHeader.java

**Comment by mjsax** [comment]
Java11 failed with https://issues.apache.org/jira/browse/KAFKA-7647
Java8 passed.

Retest this please.

**Comment by mjsax** [comment]
Call for review @guozhangwang @vvcephei @ableegoldman

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java (hunk 1 above)
Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java:92]: Thanks for the clarification.

**Review by guozhangwang — APPROVED** [review]
Thanks for the quick fix!

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.2` and `2.1`.

**Pull request merged**